### PR TITLE
@BE - create search endpoint MVP

### PR DESCRIPTION
**Pull request by dobbleXg**
@eni4sure @NdubuisiJr

### CODE REVIEW

**Comment by NdubuisiJr** [comment]
Some checks are failing fix your code

File: server/src/controllers/search.controller.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Custom Modules
+const response = require("../utils/response");
+const CustomError = require("../utils/custom-error");
+const { allGames, formatResult, formatData } = require("../utils/search_helper");
+const DatabaseConnection = require("../db/database.helper");
```
[the remark this answers is not in the document]
Comment (dobbleXg) [reply server/src/controllers/search.controller.js:5]: but that is the exact collection i intend to query, the gameRepo one

File: server/src/controllers/search.controller.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        let regex = /^[a-h][1-8]$/;
+        let keywords = ["ongoing", "games"];
+        let modifiedQuery = searchQuery.trim().toLowerCase();
+        if (keywords.includes(modifiedQuery) || regex.test(searchQuery.trim())) {
+          // get active games
+          matchedGames = await GameRepo.fetchByParameter({
+            status: 1,
+          });
+        } else {
+          // fetch all games
```
[the remark this answers is not in the document]
Comment (dobbleXg) [reply server/src/controllers/search.controller.js:31]: yes i am extracting the data in the search_helper file I created

**Comment by dobbleXg** [comment]
image
search endpoint - empty data
@NdubuisiJr

**Pull request merged**